### PR TITLE
Add basic feature engineering documentation

### PR DESCRIPTION
**Pull request by willsmithorg**
*Issue #, if available:*
1456

*Description of changes:*
Add one card to tabular documentation on feature engineering.  Mostly lists what AutoGluon does by default with a worked example.  Basics on how to add your own, and a pointer to the detailed python script in examples/tabular (already written by  innixma).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by Innixma** [comment]
Awesome! Will plan to review tomorrow.

**Comment by szha** [comment]
Job PR-1476-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1476/2/index.html

File: docs/tutorials/tabular_prediction/tabular-feature-engineering.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+Feature engineering involves taking raw tabular data and 
+
+1. converting it into a format ready for the machine learning model to read
+2. trying to enhance some columns ('features' in ML jargon) to give the ML models more information, hoping to get more accurate results.
+
+Autogluon does some of this for you.  This document describes how that works, and how you can extend it.  We describe here the default behaviour, much of which is configurable, as well as pointers to how to alter the behaviour from the default.
+
+### Column Types ###
+
+AutoGluon.Tabular recognises the following types of features, and has separate processing for them:
```
Comment (Innixma) [inline docs/tutorials/tabular_prediction/tabular-feature-engineering.md:15]: nit: autogluon.tabular or AutoGluon Tabular

File: docs/tutorials/tabular_prediction/tabular-feature-engineering.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+| Feature Type        | Example Values           |
+| :------------------ | :----------------------- |
+| boolean             | A, B                     |
+| numerical           | 1.3, 2.0, -1.6           |
+| categorical         | Red, Blue, Yellow        |
+| datetime            | 1/31/2021, Mar-31        |
+| text                | Mary had a little lamb   |
+
+In addition, other AutoGluon prediction modules recognise additional feature types, these can also be enabled in `AutoGluon.Tabular` by using the [MultiModal](tabular-multimodal.html) option.
```
Comment (Innixma) [inline docs/tutorials/tabular_prediction/tabular-feature-engineering.md:25]: nit: autogluon.tabular or AutoGluon Tabular

File: docs/tutorials/tabular_prediction/tabular-feature-engineering.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+| :------------------ | :----------------------- |
+| image               | path/image123.png        |
+
+### Column Type Detection ###
+
+- Boolean columns are any columns with only 2 unique values.
+
+- Any string columns are deemed categorical unless they are text (see below).  Some models perform better if you tell them which columns are categorical and which are continuous. 
+
+- Numeric columns are passed through without change, except to identify them as `float` or `int`.  Currently, numeric columns are not tested to determine if they are likely to be categorical.  You can force them to be treated as categorical with the Pandas syntax `.astype(category)`, see below.
```
Comment (Innixma) [inline docs/tutorials/tabular_prediction/tabular-feature-engineering.md:37]: `.astype("category")`

File: docs/tutorials/tabular_prediction/tabular-feature-engineering.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
+
+:label:`sec_feature_engineering`
```
Comment (Innixma) [inline docs/tutorials/tabular_prediction/tabular-feature-engineering.md:2]: For some reason on the side-bar, every section of this tutorial is listed. I'm unsure why this is happening, but we ideally only want it to be 1 item on the list.

The following might fix:

```
# Feature Engineering
:label:`sec_feature_engineering`
```

**Review by Innixma — COMMENTED** [review]
Looks great! This is an awesome contribution that will help users understand the nuances of AutoGluon's feature preprocessing. I've added a few comments.

File: docs/tutorials/tabular_prediction/tabular-feature-engineering.md (hunk 4 above)
Comment (willsmithorg) [reply docs/tutorials/tabular_prediction/tabular-feature-engineering.md:2]: Have updated as suggested.

File: docs/tutorials/tabular_prediction/tabular-feature-engineering.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+### Introduction ###
+
+Feature engineering involves taking raw tabular data and 
+
+1. converting it into a format ready for the machine learning model to read
+2. trying to enhance some columns ('features' in ML jargon) to give the ML models more information, hoping to get more accurate results.
+
+Autogluon does some of this for you.  This document describes how that works, and how you can extend it.  We describe here the default behaviour, much of which is configurable, as well as pointers to how to alter the behaviour from the default.
```
[the remark this answers is not in the document]
Comment (willsmithorg) [reply docs/tutorials/tabular_prediction/tabular-feature-engineering.md]: Have updated as suggested.  Sorry, not clear if I'm supposed to press 'Resolve' or you are here.

File: docs/tutorials/tabular_prediction/tabular-feature-engineering.md (hunk 1 above)
Comment (willsmithorg) [reply docs/tutorials/tabular_prediction/tabular-feature-engineering.md]: Have updated as suggested.

File: docs/tutorials/tabular_prediction/tabular-feature-engineering.md (hunk 2 above)
Comment (willsmithorg) [reply docs/tutorials/tabular_prediction/tabular-feature-engineering.md]: Have updated as suggested.

File: docs/tutorials/tabular_prediction/tabular-feature-engineering.md (hunk 3 above)
Comment (willsmithorg) [reply docs/tutorials/tabular_prediction/tabular-feature-engineering.md]: Have updated as suggested.

**Comment by szha** [comment]
Job PR-1476-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1476/3/index.html

File: docs/tutorials/tabular_prediction/tabular-feature-engineering.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Innixma) [reply docs/tutorials/tabular_prediction/tabular-feature-engineering.md]: I normally have the contributor mark as resolved after responding to the comment and pushing the fix to the PR.

**Review by Innixma — APPROVED** [review]
LGTM!

**Pull request merged**